hazelcast/src/main/java/com/hazelcast/query/impl/Indexes.java

```diff
@@ -16,13 +16,21 @@
 
 package com.hazelcast.query.impl;
 
+import com.hazelcast.config.CacheDeserializedValues;
+import com.hazelcast.config.MapConfig;
 import com.hazelcast.internal.serialization.InternalSerializationService;
+import com.hazelcast.map.impl.MapContainer;
+import com.hazelcast.map.impl.MapServiceContext;
+import com.hazelcast.map.impl.query.DefaultIndexProvider;
 import com.hazelcast.map.impl.query.IndexProvider;
```

Comment:
Seemed not expected to refer `map.impl` packages from `query.impl` package? Expecting to see those isolated.

Reply:
Have tried to refactor the code, this made constructors really messy and didn't solve the issue completely since even in the original implementation there is a reference to `com.hazelcast.map.impl.query.IndexProvider`, see https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/query/impl/Indexes.java#L20. Not sure does it worth the pains of having a constructor with 7 parameters, 3 of which are booleans. WDYT?

Reply:
if there is no easy way, i'm ok with what we have now. Maybe we can address/re-think this in another PR later.